pkg/ccl/changefeedccl/encoder.go

```diff
@@ -367,6 +370,75 @@ func (e *jsonEncoder) getTableColMap(desc catalog.TableDescriptor) catalog.Table
 	return ce.TableColMap
 }
 
+type csvEncoder struct {
+	alloc                   tree.DatumAlloc
+	virtualColumnVisibility string
+
+	buf    *bytes.Buffer
+	writer *csv.Writer
+}
+
+func newCSVEncoder(opts map[string]string) *csvEncoder {
+	newBuf := bytes.NewBuffer([]byte{})
+	newEncoder := &csvEncoder{
+		virtualColumnVisibility: opts[changefeedbase.OptVirtualColumns],
+		buf:                     newBuf,
+		writer:                  csv.NewWriter(newBuf),
+	}
+	newEncoder.writer.SkipNewline = true
+	return newEncoder
+}
+
+// EncodeKey implements the Encoder interface.
+func (e *csvEncoder) EncodeKey(_ context.Context, row encodeRow) ([]byte, error) {
+	return nil, nil
+}
+
+// EncodeValue implements the Encoder interface.
+func (e *csvEncoder) EncodeValue(_ context.Context, row encodeRow) ([]byte, error) {
+	if row.deleted {
```

Comment:
I guess that's the reason why we only support initial scan.
it's a pity, but I think it's reasonable for now.